src/components/CourseInfo/CourseInfo.tsx

```diff
@@ -8,29 +8,37 @@ import {
 } from './CourseInfo.styles';
 import { CustomTitle } from '../../common/CustomTitle/CustomTitle';
 
-import { useContext } from 'react';
-import { CoursesContext } from '../../helpers/context/courses.context';
-import { AuthorsContext } from '../../helpers/context/authors.context';
+import { useSelector } from 'react-redux';
+import { selectAuthorsList } from '../../store/authors/selectors';
+import { selectCoursesList } from '../../store/courses/selectors';
 
 import { toHoursAndMinutes } from '../../helpers/timeConvert';
 import { getArrayWithAuthors } from '../../helpers/customArrayFuncs';
 
 import { useParams } from 'react-router-dom';
 
+import { Author } from '../../store/authors/reducer';
+
 const CourseInfo = () => {
-	const { coursesList } = useContext(CoursesContext);
-	const { authorsList } = useContext(AuthorsContext);
+	const coursesList = useSelector(selectCoursesList);
+	const authorsList = useSelector(selectAuthorsList);
 
-	const { courseId } = useParams(); //or useLocation?
+	const { courseId } = useParams();
 
 	// find course to render
 	const courseToRender = coursesList.find((course) => course.id === courseId);
 	if (!courseToRender) {
-		return;
+		return <h1>Error</h1>;
 	}
 	const { id, creationDate, description, duration, title, authors } =
 		courseToRender;
 
+	const arrayWithAuthors = getArrayWithAuthors(authorsList, authors);
+	type ArrayWithAuthors = typeof arrayWithAuthors;
+	const isArrayWithAuthorsExist = (
+		arrayWithAuthors: ArrayWithAuthors
+	): arrayWithAuthors is Author[] => arrayWithAuthors !== null;
```

Comment:
new `type` is redundant here. You can do it at line 39. (because I can't see it being used somewhere else)

Reply:
BTW, your `getArrayWithAuthors` function returns Array[] | null, so you don't need to use typeof at the first place